include/threads/ThreadState.h

```diff
@@ -57,6 +56,37 @@ class Level
     std::condition_variable nowaitCv;
 };
 
+class OpenMPTask
+{
+  public:
+    OpenMPTask(faabric::Message* parentMsgIn,
+               faabric::Message& msgIn,
+               std::shared_ptr<threads::Level> nextLevelIn,
+               int threadIdxIn)
```

Comment:
Should be using smart pointers for the `parentMsg` and `msg` but that would required even more changes in this already bloated PR.